packages/data/src/cluster/ClusterIndex.ts

```diff
@@ -197,13 +206,14 @@ export function buildIndex<F>(clusterLevels: ClusterLevels): ClusterIndex<F> {
               dest: destCluster,
               count: flowCountsMapReduce.map(flow),
               aggregate: true,
+              values: [flowCountsMapReduce.map(flow)],
             };
             result.push(aggregateFlow);
             aggFlowsByKey.set(key, aggregateFlow);
           } else {
+            aggregateFlow.values.push(flowCountsMapReduce.map(flow));
```

Comment:
Why do you add `values` to flow here? It will likely significantly increase the memory use for the resulting flows data structure.

Reply:
I understand that performance issue. I try to explain why I picked this approach. Previously, you applied a map-reduce approach which means summing up every new counts added to a cluster in each iteration. This approach works fine when our aggregation function is 'sum'. 

The same can not be achieved if we wanna 'average'. To explain, say we have three edges with values of 10,5 and 8. If we apply the same approach for sum and just average them every time 'reduce' function is called, we get a different number than a real average. 
(((10+5)/2) + 8)/2 != (10+5+8)/3

This approach, despite your concern, gives the developer full flexibility on what aggregation function to be used (e.x. weighted sum, logarithmic, exponential, etc.). 

To avoid any performance loss with normal 'sum' aggregation, I change the code to use the previous 'reduce' function if getFlowAggFunc is not defined. 

Hope it makes sense and please let me know if there is any better way of achieving this flexibility with less increase in the memory use.

Reply:
I made the changes following the proposed approach. Added node aggregation feature as well.  Please let me know if they work better now. 

To be able to allow weighted aggregation I added another argument that allows to determine the weight attribute. Example would be we have a performance metric and we wanna weigthed sum it considering the volume(flow). 
getFlowMagnitude: (flow) => flow.metric_1,
 getFlowAggWeight: (flow) => flow.count,
        getFlowAggFunc: (values) => values.reduce((accumulator, curr:any) => accumulator + curr.aggvalue*curr.aggweight, 0)
        /values.reduce((acc,cur:any)=>acc + cur.aggweight,0), 

The example app seem to be working quote responsive as the memory footprint dropped dramatically.